static/src/javascripts/projects/commercial/modules/messenger/measure-ad-load.ts

```diff
@@ -20,11 +22,26 @@ top.window.postMessage(JSON.stringify(
 </script>
 * */
 
+/**
+ * Shape of data we expect to receive from creatives sending 'measure-ad-load' messages
+ */
+interface MeasureAdLoadSpecs {
+	slotId?: string;
+}
+
+const isValidMeasureAdLoadSpecs = (
+	specs: unknown,
+): specs is MeasureAdLoadSpecs =>
+	isObject(specs) && (specs.slotId === undefined || isString(specs.slotId));
```

Comment:
?

```suggestion
	isObject(specs) && (specs.slotId !== undefined && isString(specs.slotId));
```

Reply:
specs is `{ slotId?: string }`, so this was just checking whether the property is undefined or it's a string. Would your suggestion not enforce that the property is always defined and always a string?

Reply:
I was thrown by this line:

```js
isValidMeasureAdLoadSpecs(specs) ? specs.slotId : undefined
```

Implies the spec is only valid i.e. useful if `slotId` is defined otherwise we pass undefined?

Or are we just using it as a type guard?

Reply:
We're just using it as a type guard, so `specs.SlotId` may still be undefined, resulting in the same thing as an invalid specs in this case.

It looks like this might be difficult to parse at first glance, so I will think about trying to simplify it.

Reply:
Have updated it so hopefully it's a little clearer!

Reply:
That's much clearer thanks! 👍